File: world/boss_victory.lua
require "world/input"

depend("boss_room")

boss = findObject("boss")
name = Input:new()

i = ""

leaders = {}

render = function()
	window(40, 20, 0, 0)

	println("Hooray!")
	println()
	println("You beat "..boss:get("name")..", "..player:get("name").."!")

	println()
	println()

	println()
	println("Current Leaders:")
	for n, name in ipairs(leaders) do
		println(n..". "..name)
	end

	if name.done then
		println()
		println()
		println("Press any key to continue...")
	end

	cursor(2, 4)
	print("Enter your name: "..name.val)
end

update = function(input)
	if not name.done then
		name:update(input)

		if name.done then
			table.insert(leaders, 1, name.val.." beat "..boss:get("name"))
		end
	else
		move("boss_create")
	end
end

saveData = function()
	for l = 1, #leaders do
		writeln(leaders[l])
	end
end

loadData = function(input)
	for n, line in ipairs(input) do
		table.insert(leaders, line)
	end
end

File: world/boss_victory.js
var inputs = require("./input.js");
var Input = inputs.Input;

depend("boss_room");

var boss = new GameObject("boss", {});
var name = new Input();

var board = new GameObject("board", {
	level: 1
});

var insertNewWinner = function(name) {
	var update = {};
	update[board.get("level") + "_winner"] = name;
	update.level = board.get("level") + 1;

	board.set(update);
}

render = function() {
	window(40, 20, 0, 0);

	println("Hooray!");
	println("You beat " + boss.get("name") + ", " + player.get("name") + "!");

	println();
	println();

	println();
	println("Current Leaders:");
	for (var i = 1; i < board.get("level"); i ++) {
		println(i + ". " + board.get(i + "_winner"));
	}

	if (name.done) {
		println();
		println();
		println("Press any key to continue...");
	}

	cursor(2,6);
	print("Enter your name: " + name.val);

}

update = function(input) {
	if (!name.done) {
		name.update(input);

		if (name.done) {
			insertNewWinner(name.val + " beat " + boss.get("name"));
		}
	}
	else {
		move("boss_create");
	}
}
